src/controllers/movie.js

```diff
@@ -0,0 +1,204 @@
+
+const getMovies = async (req, res) => {
+  let movies
+
+  if (req.query.runtimeGt || req.query.runtimeLt) {
+    const runtimeLt = Number(req.query.runtimeLt)
+    const runtimeGt = Number(req.query.runtimeGt)
+
+    console.log(runtimeLt, runtimeGt)
+
+    if (req.query.runtimeGt && req.query.runtimeLt) {
+      // console.log('in both query')
+      movies = await prisma.movie.findMany({
+        where: {
+          runtimeMins: {
+            lt: runtimeLt,
+            gt: runtimeGt
+          }
+        },
+        include: {
+          screenings: true
+        }
+      })
+    }
+    else if (req.query.runtimeLt) {
+      // console.log('in runtimeLt')
+      movies = await prisma.movie.findMany({
+        where: {
+          runtimeMins: {
+            lt: runtimeLt
+          }
+        },
+        include: {
+          screenings: true
+        }
+      })
+    }
+    else if (req.query.runtimeGt) {
+      // console.log('in runtimeGt')
+      movies = await prisma.movie.findMany({
+        where: {
+          runtimeMins: {
+            gt: runtimeGt
+          }
+        },
+        include: {
+          screenings: true
+        }
+      })
+    }
+  }
+  else {
+    // console.log('in no query statement')
+    movies = await prisma.movie.findMany({
+      include: {
+        screenings: true
+      }
+    })
+  }
+
+  // console.log('my movies', movies)
+  res.json({ movies })
+}
+
+const createMovie = async (req, res) => {
+  const { title, runtimeMins } = req.body
+
+  if (!title || !runtimeMins ) res.status(400).json({ error: "Missing fields in the request body" })
+
+  const movie = await prisma.movie.findFirst({
+    where: { title: title }
+  })
+
+  if (movie) res.status(409).json({ error: "A movie with the provided title already exist"})
+  else {
+
+    let screenings = req.body.screenings
+    let adjustedScreenings
+
+    if (screenings) {
+      adjustedScreenings = screenings.map(screening => {
+        
+        // console.log('screening before', screening)
+        let date = screening.startsAt
+        // console.log('my date',date)
+        screening.startsAt = new Date(date)
+        // console.log(screening)
+        return screening
+      })
+    }
+
+    // console.log(adjustedScreenings)
+
+    const createdMovie = await prisma.movie.create({
+      data: !adjustedScreenings ? {
+        title,
+        runtimeMins
+      } : {
+        title,
+        runtimeMins,
+        screenings: {
+          create: adjustedScreenings
+        }
+      },
+      include: {
+        screenings: true
+      }
+    })
+    
+    console.log(createdMovie)
+
+    res.status(201).json({ movie: createdMovie })
+  }
+}
+
+const getByID = async (req, res) => {
+  
+  // let whereCondition = {}
+
+  const param = req.params.id
+
+  // if (param) {
+  //   whereCondition.id = param
+  // } else {
+  //   param = req.params.id
+  //   whereCondition.title = param
+  // }
+  // console.log('my condition', whereCondition)
```

Comment:
Commented code is usually considered "dead code" and should be removed before submitting a PR in a work environment, the git commit history exists so you have access to old code